impeller/typographer/backends/skia/typographer_context_skia.cc

```diff
@@ -33,6 +39,62 @@ namespace impeller {
 //              https://github.com/flutter/flutter/issues/114563
 constexpr auto kPadding = 2;
 
+namespace {
+
+class HostBufferAllocator : public SkBitmap::Allocator {
+ public:
+  explicit HostBufferAllocator(HostBuffer& host_buffer)
+      : host_buffer_(host_buffer) {}
+
+  BufferView TakeBufferView() {
+    buffer_view_.buffer->Flush();
+    return std::move(buffer_view_);
+  }
+
+  bool allocPixelRef(SkBitmap* bitmap) {
+    if (!bitmap) {
+      return false;
+    }
+    const SkImageInfo& info = bitmap->info();
+    if (kUnknown_SkColorType == info.colorType() || info.width() < 0 ||
+        info.height() < 0 || !info.validRowBytes(bitmap->rowBytes())) {
+      return false;
+    }
+
+    size_t required_bytes = bitmap->rowBytes() * bitmap->height();
+    BufferView buffer_view = host_buffer_.Emplace(nullptr, required_bytes,
+                                                  DefaultUniformAlignment());
+
+    struct ImpellerPixelRef final : public SkPixelRef {
```

Comment:
I am not sure why this subclass is necessary.

Reply:
Its not! removed